pkg/url/kubernetes.go

```diff
@@ -0,0 +1,108 @@
+package url
+
+import (
+	"fmt"
+	"sort"
+
+	routev1 "github.com/openshift/api/route/v1"
+	componentlabels "github.com/openshift/odo/pkg/component/labels"
+	"github.com/openshift/odo/pkg/localConfigProvider"
+	"github.com/openshift/odo/pkg/occlient"
+	"github.com/pkg/errors"
+	"k8s.io/klog"
+)
+
+type kubernetesClient struct {
+	generic
+	isRouteSupported bool
+	client           occlient.Client
+}
+
+// ListCluster lists both route and ingress based URLs from the cluster
+func (k kubernetesClient) ListCluster() (URLList, error) {
+	labelSelector := fmt.Sprintf("%v=%v", componentlabels.ComponentLabel, k.componentName)
+	klog.V(4).Infof("Listing ingresses with label selector: %v", labelSelector)
+	ingresses, err := k.client.GetKubeClient().ListIngresses(labelSelector)
+	if err != nil {
+		return URLList{}, errors.Wrap(err, "unable to list ingress")
+	}
+
+	var routes []routev1.Route
+	if k.isRouteSupported {
+		routes, err = k.client.ListRoutes(labelSelector)
+		if err != nil {
+			return URLList{}, errors.Wrap(err, "unable to list routes")
+		}
+	}
+
+	var clusterURLs []URL
+	for _, i := range ingresses {
+		clusterURL := getMachineReadableFormatIngress(i)
+		clusterURLs = append(clusterURLs, clusterURL)
+	}
+	for _, r := range routes {
+		// ignore the routes created by ingresses
+		if r.OwnerReferences != nil && r.OwnerReferences[0].Kind == "Ingress" {
+			continue
+		}
+		clusterURL := getMachineReadableFormat(r)
+		clusterURLs = append(clusterURLs, clusterURL)
+	}
+
+	return getMachineReadableFormatForList(clusterURLs), nil
+}
+
+// List lists both route/ingress based URLs and local URLs with respective states
+func (k kubernetesClient) List() (URLList, error) {
```

Comment:
Can we add more one line comments in the code? At least for the logical blocks that you have created? That would help someone to read this code in the future.

Reply:
Fixed